kubelet/datadog_checks/kubelet/summary.py

```diff
@@ -0,0 +1,149 @@
+# (C) Datadog, Inc. 2018-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+from __future__ import division
+
+from datadog_checks.base.utils.tagging import tagger
+
+from .common import tags_for_pod, tags_for_container, replace_container_rt_prefix
+
+class SummaryScraperMixin(object):
+    """
+    This class scrapes metrics from Kubelet "/stats/summary" endpoint
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(SummaryScraperMixin, self).__init__(*args, **kwargs)
+
+    def process_stats_summary(self, pod_list_utils, stats, instance_tags, main_stats_source):
+        # Reports system container metrics (node-wide)
+        self._report_system_container_metrics(stats, instance_tags)
+        # Reports POD & Container metrics. If `main_stats_source` is set, retrieve everything it can
+        # Otherwise retrieves only what we cannot get elsewhere
+        self._report_metrics(pod_list_utils, stats, instance_tags, main_stats_source)
+
+    def _report_metrics(self, pod_list_utils, stats, instance_tags, main_stats_source):
+        for pod in stats.get('pods', []):
+            pod_namespace = pod.get('podRef', {}).get('namespace')
+            pod_name = pod.get('podRef', {}).get('name')
+            pod_uid = pod.get('podRef', {}).get('uid')
+
+            if pod_namespace is None or pod_name is None or pod_uid is None:
+                self.log.warning("Kubelet returned incomplete results from '/stats/summary'")
```

Comment:
can we log the values of `pod_namespace`, `pod_name` and `pod_uid` to know which one is missing?

Reply:
Added the pod object so we know exactly what's going on. Should never happen though.